library/Freckle/App/Database.hs

```diff
@@ -137,11 +138,23 @@ runDB
   -> m a
 runDB action = do
   pool <- asks getSqlPool
+  gauge <- Stats.lookupGauge Stats.dbConnections
+  let
+    hooks = setAlterBackend defaultSqlPoolHooks $ wrapSqlBackend mempty
+    hooks' = hooks
+      { runBefore = \conn mi -> do
+          Stats.incGauge gauge
+          runBefore hooks conn mi
+      , runAfter = \conn mi -> do
+          Stats.decGauge gauge
+          runAfter hooks conn mi
+      , runOnException = \conn mi e -> do
+          Stats.decGauge gauge
+          runOnException hooks conn mi e
```

Comment:
Just for future us, do you mind adding some level of detail to the commit message?

For example, we may be curious what was wrong before and how this fixes it, and I'd be interested in a link to the hooks docs. There seems to be an assumption here that the "after" hook only runs on success, requiring we also hook into "on exception". Is that documented anywhere you can link?

Reply:
of course, was intending on doing this; at the time I wasn't totally sure that I was going to use this technique, although, actually your 

I actually added it as a comment, as I thought the area tricky enough that a comment would stand out more.